zerver/actions/message_edit.py

```diff
@@ -824,7 +824,17 @@ def user_info(um: UserMessage) -> Dict[str, Any]:
 
     send_event(user_profile.realm, event, users_to_be_notified)
 
-    if len(changed_messages) > 0 and new_stream is not None and stream_being_edited is not None:
+    is_topic_changed = False
+    if topic_name is not None:
+        cleared_topic_name = topic_name.replace("✔", "").lstrip()
+        cleared_orig_topic_name = orig_topic_name.replace("✔", "").lstrip()
+        is_topic_changed = cleared_topic_name != cleared_orig_topic_name
+
+    if (
+        len(changed_messages) > 0
+        and (new_stream is not None or is_topic_changed)
+        and stream_being_edited is not None
+    ):
```

Comment:
The intent here is to avoid sending both moved message and resolved topic notifications, right? I think a better approach would be to (1) reorder the checks so we do the resolved topic check first, (2) have `maybe_send_resolve_topic_notifications` return whether it moved anything, and (3) then check that value in this block, rather than reimplementing logic to check whether we'll send resolve topic notifications.